server/src/main/java/org/apache/druid/server/coordinator/duty/KillUnusedSegments.java

```diff
@@ -191,38 +207,43 @@ private int killUnusedSegments(
     if (0 < availableKillTaskSlots && !CollectionUtils.isNullOrEmpty(dataSourcesToKill)) {
       for (String dataSource : dataSourcesToKill) {
         if (submittedTasks >= availableKillTaskSlots) {
-          log.debug(StringUtils.format(
+          log.info(StringUtils.format(
               "Submitted [%d] kill tasks and reached kill task slot limit [%d]. Will resume "
               + "on the next coordinator cycle.", submittedTasks, availableKillTaskSlots));
           break;
         }
-        final Interval intervalToKill = findIntervalForKill(dataSource);
+        final DateTime maxUsedFlagLastUpdatedTime = DateTimes.nowUtc().minus(bufferPeriod);
+        final Interval intervalToKill = findIntervalForKill(dataSource, maxUsedFlagLastUpdatedTime);
         if (intervalToKill == null) {
           datasourceToLastKillIntervalEnd.remove(dataSource);
           continue;
         }
 
         try {
-          FutureUtils.getUnchecked(overlordClient.runKillTask(
-              TASK_ID_PREFIX,
-              dataSource,
-              intervalToKill,
-              maxSegmentsToKill
-          ), true);
+          FutureUtils.getUnchecked(
+              overlordClient.runKillTask(
+                  TASK_ID_PREFIX,
+                  dataSource,
+                  intervalToKill,
+                  maxSegmentsToKill,
+                  maxUsedFlagLastUpdatedTime
+              ),
+              true
+          );
           ++submittedTasks;
           datasourceToLastKillIntervalEnd.put(dataSource, intervalToKill.getEnd());
         }
         catch (Exception ex) {
-          log.error(ex, "Failed to submit kill task for dataSource [%s]", dataSource);
+          log.error(ex, "Failed to submit kill task for dataSource[%s] in interval[%s]", dataSource, intervalToKill);
           if (Thread.currentThread().isInterrupted()) {
-            log.warn("skipping kill task scheduling because thread is interrupted.");
+            log.warn("Skipping kill task scheduling because thread is interrupted.");
             break;
           }
         }
       }
     }
 
-    if (log.isDebugEnabled()) {
+    if (!log.isDebugEnabled()) {
```

Comment:
Why change this condition? I think we only want to compute the subList in the log message here if debug logging is enabled?

Reply:
Good catch. Inverted this condition for testing, will put it back